src/sentry/llm/providers/base.py

```diff
@@ -0,0 +1,46 @@
+from typing import Any
+
+from sentry.llm.exceptions import InvalidModelError, InvalidProviderError
+from sentry.utils.services import Service
+
+
+class LlmModelBase(Service):
+    def __init__(self, provider_config: dict[str, Any]) -> None:
+        self.provider_config = provider_config
+
+    def complete_prompt(
+        self,
+        *,
+        usecase_config: dict[str, Any],
```

Comment:
this should probably be either a real type or a `TypedDict` -- especially in brand new code we should not be using `dict[str, Any]`

Reply:
i tried typing with a typed dict and it proved to be annoying as each provider can have different options. can try again

Reply:
Added a TypedDict with more narrow types, with options being `dict[str,str]`. doesn't seem like narrowing the specific types for each option set further from there is easy